ballerina/tests/test.bal

```diff
@@ -0,0 +1,297 @@
+
+// Test to get a list of all tickets
+@test:Config{
+    groups: ["live_tests"]
+}
+
+isolated function getTickets() returns error? {
+    CollectionResponseSimplePublicObjectWithAssociationsForwardPaging response2 = check HubSpotClient->/.get();
+    if (response2.results is SimplePublicObject[]) {
+        test:assertTrue(response2.results.length()>0, "No tickets found");
+    }
+}
+
+// Test case to create a new ticket
+@test:Config{
+    groups: ["live_tests"]
+}
+function createTicket() returns error? {
+    SimplePublicObjectInputForCreate payload = {
+        properties: {
+            "hs_pipeline": "0",
+            "hs_pipeline_stage": "1",
+            "hs_ticket_priority": "LOW",
+            "subject": "Auto generated Issue Report"
+        }
+    };
+    SimplePublicObject|error response5 = check HubSpotClient->/.post(payload);
+    if (response5 is SimplePublicObject) {
+        test:assertTrue(response5?.properties["hs_object_id"] != null, "Response is not a valid");
+        test:assertTrue(response5?.properties["subject"] == "Auto generated Issue Report", "Response is not a valid");
+        test:assertTrue(response5?.properties["hs_ticket_priority"] == "LOW", "Response is not a valid");
+        test:assertTrue(response5.properties["hs_pipeline"] == "0", "Response is not a valid");
+        string? createdTicketId = response5.properties["hs_object_id"];
+        test:assertTrue(createdTicketId != null, "Response does not contain a valid ticket ID");
+        ticketId = createdTicketId ?: "";
+    }
+    else{
+        test:assertFail("Response is not valid");
+    }
+}
+
+// Test case to get the ticket by id
+@test:Config{
+    groups: ["live_tests"],
+    dependsOn: [createTicket]
+}
+
+function getTicketById() returns error? {
+    SimplePublicObjectWithAssociations response3 = check HubSpotClient->/[ticketId].get();
+    test:assertTrue(response3?.properties["subject"]=="Auto generated Issue Report", "Response is not correct");
+    test:assertTrue(response3?.properties["hs_ticket_priority"]=="LOW", "Response is not correct");
+    test:assertTrue(response3?.properties["hs_pipeline"]=="0", "Response is not correct");
+}
+
+// Test to check Update the ticket by id
+@test:Config{
+    groups: ["live_tests"],
+    dependsOn: [getTicketById]
+}
+
+function updateTicketById() returns error? {
+    SimplePublicObjectInput payload = {
+        properties: {
+        "subject": "Updated Bug Fix",
+        "hs_ticket_priority": "HIGH",
+        "hs_pipeline": "0"
+        }
+    };
+    SimplePublicObject response4 = check HubSpotClient->/[ticketId].patch(payload);
+    test:assertTrue(response4?.properties["subject"]=="Updated Bug Fix", "Response is not a Updated");
+    test:assertTrue(response4?.properties["hs_ticket_priority"]=="HIGH", "Response is not a Updated");
+    test:assertTrue(response4?.properties["hs_pipeline"]=="0", "Response is not a Updated");
```

Comment:
All the failure messages are same in this case, which is redundant. We can either remove all of them, or use specific failure messages.

```suggestion
    test:assertEquals(response4?.properties["subject"], "Updated Bug Fix", "Response is not a Updated");
    test:assertEquals(response4?.properties["hs_ticket_priority"], "HIGH", "Response is not a Updated");
    test:assertEquals(response4?.properties["hs_pipeline"], "0", "Response is not a Updated");
```